Library/Homebrew/cmd/update.rb

```diff
@@ -36,13 +45,27 @@ def update
         rescue
           onoe "Failed to update tap: #{tap}"
         else
-          report.update(updater.report) do |key, oldval, newval|
-            oldval.concat(newval)
+          updater.report
```

Comment:
I don't think the actual output text should be inside the thread. Might need wider refactoring.